### PR TITLE
Update dependencies, linting, readonlyness

### PR DESCRIPTION
**Pull request by danielnixon**

### CODE REVIEW

File: cli/.eslintrc.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ module.exports = {
     "total-functions",
   ],
   rules: {
+    "functional/prefer-immutable-types": ["error", { "enforcement": "ReadonlyShallow" }],
```
Comment (danielnixon) [inline cli/.eslintrc.js:23]: This PR gets us up to `ReadonlyShallow`. I might go for `ReadonlyDeep` in a subsequent PR. I probably won't bother with truly `Immutable`.

File: lib/src/services/issue_checks.ts [hunk 2]
```diff
@@ -14,6 +14,8 @@ import {
 } from "date-fns";
 import { ReadonlyNonEmptyArray } from "fp-ts/lib/ReadonlyNonEmptyArray";
 
+// TODO unclear why ReadonlyNonEmptyArray is not judged to be Immutable
```
Comment (danielnixon) [inline lib/src/services/issue_checks.ts:17]: This is a puzzle I have yet to solve

File: cli/.eslintrc.js (hunk 1 above)
Comment (danielnixon) [reply cli/.eslintrc.js]: I went all the way to immutable

**Comment by danielnixon** [comment]
Looks like one of our eslint plugins doesn't like node 15. Suggest we update CI to use 16 and 18.

**Comment by danielnixon** [comment]
https://github.com/agiledigital/jiralint/blob/main/.github/workflows/build-test-release.yml#L12 versus https://github.com/eslint-functional/eslint-plugin-functional/blob/main/package.json#L162

**Comment by dspasojevic** [comment]
I wonder what semantic release will make of these commit messages.

**Comment by danielnixon** [comment]
> I wonder what semantic release will make of these commit messages.

Sorry 🙈

**Pull request merged**